cache/src/main/java/io/envoyproxy/controlplane/cache/SimpleCache.java

```diff
@@ -196,25 +196,30 @@ public void setSnapshot(T group, Snapshot snapshot) {
       if (status == null) {
         return;
       }
+      for (String typeUrl : Resources.TYPE_URLS) {
+        status.watchesRemoveIf((id, watch) -> {
+          if (!watch.request().getTypeUrl().equals(typeUrl)) {
```

Comment:
Does it work with ADS? The typeUrl of ADS is an empty string, so I think this will be always false.

Reply:
Oh, I'm sorry, this is on a different level. Callbacks receive empty string on ADS, the type url is present in `DiscoveryRequest`